src/ControlSystem/ControlErrors/Size/Error.hpp

```diff
@@ -24,6 +24,28 @@ class ZeroCrossingPredictor;
 /// \endcond
 
 namespace control_system::size {
+/*!
+ * \brief A simple struct to hold diagnostic information about computing the
+ * size control error.
+ */
+struct ErrorDiagnostics {
+  size_t state_number;
+  double lambda_00;
+  double dt_lambda_00;
+  double horizon_00;
+  double dt_horizon_00;
+  double control_error_delta_r;
+  double min_char_speed;
+  double min_comoving_char_speed;
+  double char_speed_crossing_time;
+  double comoving_char_speed_crossing_time;
+  double delta_r_crossing_time;
+  double control_error;
```

Comment:
[optional] Stylistically, if this struct is the control error + diagnostic information, it might be clearer to have the control error at the start or end instead of in the middle of the diagnostics.